From df103442f7dd2de72e74c1dfb4df0b157171e3cd Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Wed, 14 Mar 2018 17:53:05 +0800
Subject: [PATCH 3510/5242] MLK-17804-1: dts: fix ISI and MIPI CSI power
 domain can't enter lp mode

commit  fa646b90cba50761067e1b2b69ba8046240f641a from
https://source.codeaurora.org/external/imx/linux-imx.git

MIPI CSI power domain can not enter low power mode because gpio of
mipi csi module occupy the resource, but camera sensor always use
the gpio, so making gpio of gpio3 controlloer replace mipi csi gpio
to control sensor EN and RESET pins.

Reviewed-by: Sandor.yu <sandor.yu@nxp.com>
Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit d1a018e1fca614f963c17c155f1a42abcce19bdf)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8qxp-mek-mipi-csi2.dts    |    6 ++++--
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts  |    5 -----
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-mipi-csi2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-mipi-csi2.dts
index 459d302..e8460dc 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-mipi-csi2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-mipi-csi2.dts
@@ -19,6 +19,8 @@
 		pinctrl_mipi_csi0: mipicsi0grp{
 			fsl,pins = <
 				SC_P_MIPI_CSI0_MCLK_OUT_MIPI_CSI0_ACM_MCLK_OUT	0xC0000041
+				SC_P_MIPI_CSI0_GPIO0_01_LSIO_GPIO3_IO07		0xC0000041
+				SC_P_MIPI_CSI0_GPIO0_00_LSIO_GPIO3_IO08		0xC0000041
 			>;
 		};
 	};
@@ -52,8 +54,8 @@
 		clocks = <&clk IMX8QXP_CLK_DUMMY>;
 		clock-names = "csi_mclk";
 		csi_id = <0>;
-		pwn-gpios = <&gpio0_mipi_csi0 1 GPIO_ACTIVE_LOW>;
-		rst-gpios = <&gpio0_mipi_csi0 0 GPIO_ACTIVE_HIGH>;
+		pwn-gpios = <&gpio3 7 GPIO_ACTIVE_LOW>;
+		rst-gpios = <&gpio3 8 GPIO_ACTIVE_HIGH>;
 		mclk = <24000000>;
 		mclk_source = <0>;
 		mipi_csi;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index 55ef477..e091fb2 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -829,11 +829,6 @@
 	status = "okay";
 };
 
-&gpio0_mipi_csi0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_mipi_csi0_gpio>;
-};
-
 &mipi_csi_0 {
 	#address-cells = <1>;
 	#size-cells = <0>;
-- 
1.7.9.5

